### DISCUSSION TITLE
All-zero listening ports in the RabbitMQ container

### DISCUSSION THREAD

**Discussion by dormanze** [rabbitmq/rabbitmq-server#8973] [Questions] [answered]
By chance, I found that the RabbitMQ container has a resident all-zero listening port. I don't know what the function of this port is. It is a UDP protocol. I don't specify this port in the configuration, and I don't find this port in the rabbitmq source code. Oddly enough, I didn't find this port in my other containers, and when I restarted the container, it disappeared, and I wanted to know what this port was for.
rabbitmq version: 3.11.9
erlang version: 25.2.2
![image](https://github.com/rabbitmq/rabbitmq-server/assets/37854724/01f8ba6a-701f-4158-9cbc-082758612349)

**Comment by michaelklishin** [comment, marked as answer]
The output suggests the port is used by a RabbitMQ node.

RabbitMQ does not use UDP except for one (non-tier 1, that is, not shipped with RabbitMQ) exchange type plugin, [rabbitmq-udp-exchange](https://github.com/rabbitmq/rabbitmq-udp-exchange). All the protocols RabbitMQ supports are TCP-based.

See `rabbitmq-plugins list` or node logs to find out what plugins are enabled on the node.

The [community Docker image for RabbitMQ](https://github.com/docker-library/rabbitmq) does not install and enable that plugin. If I were to guess, neither should be Bitnami and the VMware RabbitMQ images.

**Comment by dormanze** [comment]
Thank you for your reply.
At first, I suspected that the port was enabled by a service on the VM. However, when I saw the process name beam.smp, I suspected that the port was enabled by RabbitMQ or Erlang.
